### PR TITLE
Support docker compose v2

### PR DESCRIPTION
**Pull request by Julien-Eyraud**
## Type of change
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Support for both docker compose versions:
- the legacy: docker-compose
- the v2 as a docker plugin: docker compose

## Code changes
A small change in the installations script to choose between the two commands depending on the system setup.

* **install.sh:** Check if legacy or v2 is installed and declare the command to use as a variable.
* **run.sh:** Check if legacy or v2 is installed and declare the command to use as a variable. Use the docker compose command variable defined in setup

## Testing requirements
Tests all commands:
- install
- start/restart
- stop
- update
- rebuild

## Before you submit
- [x ] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [x ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by Hinton** [comment]
@joseph-flinn Any thoughts? I suspect we'll want to do the same change for powershell.

Also not sure what we'll do if there are inconsistencies between the two versions of docker compose.

https://github.com/docker/compose-switch could also be an alternative.

**Comment by Julien-Eyraud** [comment]
I'm not familiar with powershell that's why I didn't provide the support in this PR. For now I'm not aware of any inconsistencies between the two versions. It suppose to be a dropping replacement for the v1. More on the compatibility here https://docs.docker.com/compose/cli-command-compatibility/.

Using compose-switch require another component to be installed that could be avoid IMHO.

File: scripts/bitwarden.sh [hunk 1]
```diff
@@ -40,7 +46,7 @@ KEYCONNECTORVERSION="1.0.0"
 
 echo "bitwarden.sh version $COREVERSION"
 docker --version
-docker-compose --version
+$dccmd --version
```
Comment (joseph-flinn) [inline scripts/bitwarden.sh:49]: `--version` is not supported by `docker compose`

```suggestion
if [[ "$dccmd" == "docker compose" ]]; then
    $dccmd version
else
    $dccmd --version
fi
```

File: scripts/bitwarden.sh [hunk 2]
```diff
@@ -29,6 +29,12 @@ if [ $# -eq 2 ]
 then
     OUTPUT=$2
 fi
+if command -v docker-compose &> /dev/null
+then
+  dccmd='docker-compose'
+else
+  dccmd='docker compose'
+fi
```
Comment (joseph-flinn) [inline scripts/bitwarden.sh:37]: To follow our current bash coding standards:

```suggestion
then
    dccmd='docker-compose'
else
    dccmd='docker compose'
fi
```

File: scripts/run.sh [hunk 3]
```diff
@@ -2,6 +2,12 @@
 set -e
 
 # Setup
+if command -v docker-compose &> /dev/null
+then
+  dccmd='docker-compose'
+else
+  dccmd='docker compose'
+fi
```
Comment (joseph-flinn) [inline scripts/run.sh:10]: To follow our bash coding standards:
```suggestion
then
    dccmd='docker-compose'
else
    dccmd='docker compose'
fi
```

**Review by joseph-flinn — CHANGES_REQUESTED** [review]
@Julien-Eyraud Thanks for your contribution! I apologize for the delay in taking a look at your PR. I've left some comments after testing the changes

**Comment by Julien-Eyraud** [comment]
@joseph-flinn thanks for the review I took your changes into account.

**Review by joseph-flinn — APPROVED** [review]

**Comment by joseph-flinn** [comment]
@Julien-Eyraud Great! We will get this in with the release this week (targeting Thursday).

**Review by joseph-flinn — APPROVED** [review]

**Pull request merged**